components/script/dom/document.rs

```diff
@@ -592,6 +596,24 @@ impl Document {
         }
     }
 
+    /// Add a focusable element to this document's ordering focus list.
+    pub fn add_focusable_element(&self, element: &Element) {
+        self.focus_list.borrow_mut().push(JS::from_ref(element));
+    }
+
+    /// Removes a focusable element from this document's ordering focus list.
+    pub fn remove_focusable_element(&self, element: &Element) {
+        let mut list = self.focus_list.borrow_mut();
+
+        let index = list.iter().position(|item| {
+            &**item == element
+        });
+
+        if let Some(index) = index {
+            list.remove(index);
+        }
```

Comment:
We will need to ensure that the current focused index is updated to account for this change.